lib/http_aws_sigv4.c

```diff
@@ -154,6 +154,66 @@ static int compare_header_names(const char *a, const char *b)
   return cmp;
 }
 
+/* merge duplicate header definitions by comma delimiting their values
+   in the order defined the headers are defined */
+static CURLcode merge_duplicate_headers(struct curl_slist *head)
+{
+  struct curl_slist *curr = head;
+
+  while(curr) {
+    struct curl_slist *next = curr->next;
+    if(!next) {
+      break;
+    }
+
+    if(compare_header_names(curr->data, next->data) == 0) {
+      char *colon_next;
+      char *merged;
+      char *val_next;
+      size_t curr_len;
+      size_t val_next_len;
+      size_t new_len;
+
+      colon_next = strchr(next->data, ':');
```

Comment:
maybe add a test that provides a header without a colon? Seems likely to be able to happen and an assert is not going to catch that in a release build.

Reply:
Added to test